### PR TITLE
[algo] fix grpo passk

### LINKED ISSUE #585: GRPO_PASSK implementation mismatch with paper causes training collapse
https://github.com/hiyouga/EasyR1/blob/4d13ebe55ad75230305e58a3e9cf48cd58042258/verl/trainer/core_algos.py#L220

#### Description
The `compute_grpo_passk_outcome_advantage` function is documented to assign a non-zero advantage **only to the best response per group** (calculated as `r_max - r_second_max`), with all other responses in the group set to 0. However, the current implementation fails to zero out the scores of non-best responses, leading to incorrect advantage values and eventual training collapse.

#### Impact
This bug breaks the core GRPO-Pass@k advantage formulation described in the paper (https://arxiv.org/abs/2503.19595) and leads to non-deterministic training collapse, as the advantage signal is polluted by non-zero values for non-optimal responses.

#### Verified Fix
I modified the implementation to strictly enforce "only the best response has non-zero advantage" (all others = 0), and training no longer collapses with this fix. Below is the corrected code and key changes:
```
@register_adv_estimator(AdvantageEstimator.GRPO_PASSK)
def compute_grpo_passk_outcome_advantage(
    token_level_rewards: torch.Tensor, response_mask: torch.Tensor, index: List, eps: float = 1e-6, **kwargs
) -> tuple[torch.Tensor, torch.Tensor]:
    """
    Compute advantage for Pass@k using a GRPO-style outcome reward formulation.
    Only the best response per group gets a non-zero advantage: r_max - r_second_max.

    Implemented as described in https://arxiv.org/abs/2503.19595.

    Args:
        token_level_rewards: `(torch.Tensor)`
            shape: (bs, response_length)
        response_mask: `(torch.Tensor)`
            shape: (bs, response_length)
        index: `(torch.Tensor)`
            shape: (bs,)
        eps: `(float)`
            epsilon value to avoid division by zero

    Returns:
        advantages: `(torch.Tensor)`
            shape: (bs, response_length)
        returns: `(torch.Tensor)`
            shape: (bs, response_length)

    """
    scores = token_level_rewards.sum(dim=-1)
    advantages = torch.zeros_like(scores)
    id2score = defaultdict(list)
    id2indices = defaultdict(list)

    bsz = scores.shape[0]
    for i in range(bsz):
        id2score[index[i]].append(scores[i])
        id2indices[index[i]].append(i)

    for idx in id2score:
        assert len(id2score[idx]) > 1, "GRPO needs rollout.n > 1."
        rewards = torch.stack(id2score[idx])
        topk, topk_idx = torch.topk(rewards, k=2)
        r_max, r_second_max = topk[0], topk[1]
        i_max = id2indices[idx][topk_idx[0]]
        advantages[i_max] = (r_max - r_second_max) / (torch.std(rewards) + eps)

    returns = advantages.unsqueeze(-1) * response_mask
    return returns, returns
```

### PR DESCRIPTION
**Pull request by hiyouga**
fixes #585

### CODE REVIEW

**Comment by hiyouga** [comment]
@Kuangdd01
